src/app/visa/c.spec.ts

```diff
@@ -0,0 +1,630 @@
+            })
+
+            it('1 year of experience', () => {
+                const checklist = simulationWithCriteriaC([
+                    professionalCareerWith({ yearsOfExperience: 1 }),
+                ])
+
+                const { matches, points } = calculatePoints(checklist)
+
+                expect(points).toBe(0)
+                expect(matches.map(m => m.id).sort()).toEqual([])
+            })
+        })
+
+        describe('annual salary', () => {
+            it('2.9 million', () => {
+                const checklist = simulationWithCriteriaC([
+                    annualSalaryOf(2_999_999), // 0 points
+                ])
+
+                expect(() => {
+                    calculatePoints(checklist)
+                }).toThrowError(errorMessages.salaryTooLow)
+            })
+            it('10 million', () => {
+                const checklist = simulationWithCriteriaC([
+                    annualSalaryOf(10_000_000), // 10 points
+                ])
+
+                const { matches, points } = calculatePoints(checklist)
+
+                expect(points).toBe(10)
+                expect(matches.map(m => m.id).sort()).toEqual(['10m_or_more'])
+            })
+            it('15 million', () => {
+                const checklist = simulationWithCriteriaC([
+                    annualSalaryOf(15_000_000), // 20 points
+                ])
+
+                const { matches, points } = calculatePoints(checklist)
+
+                expect(points).toBe(20)
+                expect(matches.map(m => m.id).sort()).toEqual(['15m_or_more'])
+            })
+            it('20 million', () => {
+                const checklist = simulationWithCriteriaC([
+                    annualSalaryOf(20_000_000), // 30 points
+                ])
+
+                const { matches, points } = calculatePoints(checklist)
+
+                expect(points).toBe(30)
+                expect(matches.map(m => m.id).sort()).toEqual(['20m_or_more'])
+            })
+            it('25 million', () => {
+                const checklist = simulationWithCriteriaC([
+                    annualSalaryOf(25_000_000), // 40 points
+                ])
+
+                const { matches, points } = calculatePoints(checklist)
+
+                expect(points).toBe(40)
+                expect(matches.map(m => m.id).sort()).toEqual(['25m_or_more'])
+            })
+            it('30 million', () => {
+                const checklist = simulationWithCriteriaC([
+                    annualSalaryOf(30_000_000), // 50 points
+                ])
+
+                const { matches, points } = calculatePoints(checklist)
+
+                expect(points).toBe(50)
+                expect(matches.map(m => m.id).sort()).toEqual(['30m_or_more'])
+            })
+        })
+
+        describe('position in company', () => {
+            it('representative director', () => {
+                const checklist = simulationWithCriteriaC([
```

Comment:
For test cases it's good to follow the Arrange-Act-Assert format as described [here](https://automationpanda.com/2020/07/07/arrange-act-assert-a-pattern-for-writing-good-tests/).

Having it arranged that way in blocks makes it easier for others reading the tests to understand how they work and allows for efficient scanning.

Reply:
isn't this already following the  Arrange-Act-Assert  format that you sent>

Reply:
It's in that order but not in separate blocks, which makes it harder to scan the test visually.